LibraBFT/Abstract/BFT.agda

```diff
@@ -35,13 +38,22 @@ module LibraBFT.Abstract.BFT
  Meta-dishonest? :  ∀ (m : Member) → Dec (Meta-Dishonest-PK (getPubKey m))
  Meta-dishonest? m = Meta-DishonestPK? (getPubKey m)
 
- module _  (bft-assumption : ∀ {xs : List Member} → IsSorted _<Fin_ xs
-                           → length (List-filter Meta-dishonest? xs) ≤ bizF)
+ CombinedPower : List Member → ℕ
+ CombinedPower xs = sum (List-map votPower xs)
+
+ --TODO-1 : replace IsSorted with allDistict
+ module _  (totalVotPower  : N ≡ CombinedPower (List-tabulate id))
+           (bft-assumption : ∀ {xs : List Member}
+                           → IsSorted _<Fin_ xs
+                           → CombinedPower (List-filter Meta-dishonest? xs) ≤ bizF)
    where
 
-   _∈?_ : (x : Member) → (xs : List Member) → Dec (Any (x ≡_) xs)
+   _∈?_ : ∀ {n} (x : Fin n) → (xs : List (Fin n)) → Dec (Any (x ≡_) xs)
    x ∈? xs = Any-any (x ≟Fin_) xs
 
+   participants : List Member
+   participants = List-tabulate id
+
    -- TODO-2 : Many of these lemmas can be generalized for any list or any
    -- IsSorted list of Fin. Perhaps establish a Lemmas.FinProps module.
    intersectElem : List Member → Member → List Member
```

Comment:
And another comment about `intDiffElem` and related properties that use `Diff` in the name.  I find this awkward because `Diff` makes me think of set difference, whereas what you mean is that the elements are different.  Maybe `Distinct` is better?  Or maybe `Sorted` (which would change when/if someone addresses the `TODO` about generalizing the assumptions to depend on `allDistinct` instead of `Sorted`); that would be consistent with `union-sorted`.

Reply:
Another style note, I think it's preferable to use camel case, like `unionSorted` rather than hyphens.  I'm not sure we've been consistent about this everywhere, but we should try to be, so perhaps you could set a good example?

Reply:
So many complains about `intDiffElem`: I just deleted it :) First I made your suggestions but then I noticed that actually we didn't need it where it was being used, I proved that lemma in a simpler way, without needing it. 

Regarding the `union-sorted`, I changed, but actually we are not being consistent with that I think. I can change it through the file if you want

Reply:
I think it would be nice to move in the direction of consistency, if you have time.  But as you say, we are not consistent elsewhere (yet) anyway.

Reply:
I won't bug you more about this for now, we're inconsistent all over the place anyway and haven't come up with clear rules to follow.